R/nm-join.R

```diff
@@ -109,6 +109,12 @@ nm_join <- function(
     stop(glue("couldn't find `.join_col` {.join_col} in data with cols: {paste(names(.d), collapse = ', ')}"))
   }
 
+  if(.d[.join_col] %>% anyDuplicated() != 0)
+  {
+   dup_row <- .d[.join_col][duplicated( .d[.join_col]) %>% which(),]
+   stop(glue("Duplicate rows in data in {.join_col}: {dup_row}"))
```

Comment:
@barrettk @rymarinelli Can you check what is the output for this in case there are thousands of rows that are duplicated (likely if you get this far). I'm not sure it'll be pretty or helpful output.

Reply:
Here's output for 100 duplicates in `.new_table$NUM`; from my tests on simple data sets, it seems likely that we'll have 1000s of duplicates.

```r
Browse[2]> .new_table$NUM[1:100] <- 10000001
Browse[2]> 
debug: dup_row <- .new_table[.join_col][duplicated(.new_table[.join_col]) %>% 
    which(), ]
Browse[2]> 
debug: stop(glue("Duplicate rows in {.join_col}: {dup_row}"))
Browse[2]>     stop(glue("Duplicate rows in {.join_col}: {dup_row}"))
Error in drop_dups(tab, .d, .join_col, .n) : 
  Duplicate rows in NUM: c(10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 
10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000001, 10000
Browse[2]> 
```